PROJECT/ProjectV1_1/Modules/Webserver_VMSS_AppGw.bicep

```diff
@@ -0,0 +1,710 @@
+          }
+        }
+      }
+    ]
+    httpListeners: [
+      {
+        name: 'HttpListener'
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/applicationGateways/frontendIPConfigurations/', applicationGatewayName, 'appGatewayFrontendIP')
+          }
+          frontendPort: {
+            id: resourceId('Microsoft.Network/applicationGateways/frontendPorts/', applicationGatewayName, 'FrontEndhttpPort80')
+          }
+          protocol: 'Http'
+        }
+      }
+      {
+        name: 'HttpsListener'
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/applicationGateways/frontendIPConfigurations/', applicationGatewayName, 'appGatewayFrontendIP')
+          }
+          frontendPort: {
+            id: resourceId('Microsoft.Network/applicationGateways/frontendPorts/', applicationGatewayName, 'FrontEndhttpsPort443')
+          }
+          protocol: 'Https'
+          sslCertificate: {
+            id: resourceId('Microsoft.Network/applicationGateways/sslCertificates/', applicationGatewayName, '${applicationGatewayName}SslCert')
+          }
+        }
+      }
+    ]
+    requestRoutingRules: [
+      {
+        name: 'httpPort80rule'
+        properties: {
+          ruleType: 'Basic'
+          httpListener: {
+            id: resourceId('Microsoft.Network/applicationGateways/httpListeners', applicationGatewayName, 'HttpListener')
+          }
+          redirectConfiguration: {
+            id: resourceId('Microsoft.Network/applicationGateways/redirectConfigurations/', applicationGatewayName, 'httpToHttps')
+          }
+        }
+      }
+      {
+        name: 'httpsPort443rule'
+        properties: {
+          ruleType: 'Basic'
+          httpListener: {
+            id: resourceId('Microsoft.Network/applicationGateways/httpListeners', applicationGatewayName, 'HttpsListener')
+          }
+          backendAddressPool: {
+            id: resourceId('Microsoft.Network/applicationGateways/backendAddressPools', applicationGatewayName, 'appGatewayBackendPool')
+          }
+          backendHttpSettings: {
+            id: resourceId('Microsoft.Network/applicationGateways/backendHttpSettingsCollection', applicationGatewayName, 'HTTPBackEnd')
+          }
+        }
+      }
+    ]
+    redirectConfigurations: [
+      {
+        name: 'httpToHttps'
+        properties: {
+          redirectType: 'Permanent'
+          targetListener: {
+            id: resourceId('Microsoft.Network/applicationGateways/httpListeners/', applicationGatewayName, 'HttpsListener')
+          }
+          includePath: true
+          includeQueryString: true
+          requestRoutingRules: [
+            {
+              id: resourceId('Microsoft.Network/applicationGateways/requestRoutingRules/', applicationGatewayName, 'httpPort80rule')
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+
+
+
+
+
+
```

Comment:
This is where the Scale Set starts. It runs One machine and expands when necessary to max 3 instances.